dependency.go

```diff
@@ -0,0 +1,171 @@
+package testcontainers
+
+import (
+	"context"
+	"fmt"
+	"github.com/docker/docker/api/types/container"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"golang.org/x/exp/slices"
+)
+
+// ContainerDependency represents a reliance that a container has on another container.
+type ContainerDependency struct {
+	Request ContainerRequest
+	EnvKey  string
+	// CallbackFunc is called after the dependency container is started.
+	CallbackFunc func(Container)
+	// KeepAlive determines whether the dependency should be kept alive after the parent container is terminated.
+	KeepAlive bool
+}
+
+// NewContainerDependency can be used to define a dependency and the environment variable that
+// will be used to pass the DNS name to the parent container.
+func NewContainerDependency(containerRequest ContainerRequest, envKey string) ContainerDependency {
+	return ContainerDependency{
+		Request:      containerRequest,
+		EnvKey:       envKey,
```

Comment:
I think this is specific to an use case (setting an environment variable with the IP address of the dependency) and can be implemented by the call-back function, so in think shouldn't be part of the generic API.